src/miniq.js

```diff
@@ -0,0 +1,15 @@
+(function () {
+  'use strict';
+
+  Promise.delay = (ms) => new Promise((resolve) => setTimeout(resolve, ms))
+
+  Promise.defer = () => {
+    let resolve, reject;
+    const promise = new Promise((_resolve, _reject) => {
+      resolve = _resolve;
+      reject = _reject;
+    })
+    return {promise, resolve, reject};
```

Comment:
I like that you avoided assigning to fields of the deferred.

Reply:
When in Rome.